### PR TITLE
Poll consumer while there are messages to consume

### PR DESCRIPTION
**Pull request by jlabedo**
Add `--finishWhenNoMessages` option when running asynchronous endpoints. The execution will be stopped when there are no messages to handle

### CODE REVIEW

File: packages/Ecotone/tests/Messaging/Unit/Endpoint/Poller/PollingConsumerBuilderTest.php [hunk 1]
```diff
@@ -393,6 +395,60 @@ public function test_throwing_exception_and_requeing_when_stop_on_failure_define
         $this->assertNull($errorChannel->receive());
     }
 
+    public function test_finish_when_no_messages(): void
+    {
+        $inputChannelName = 'inputChannel';
+        $inputChannel = new class implements PollableChannel {
```
Comment (dgafka) [inline packages/Ecotone/tests/Messaging/Unit/Endpoint/Poller/PollingConsumerBuilderTest.php:401]: You can switch this to `QueueChannel::create()`, as this is default In Memory Pollable implementation :)

File: packages/Ecotone/tests/Messaging/Unit/Endpoint/Poller/PollingConsumerBuilderTest.php [hunk 2]
```diff
@@ -393,6 +395,60 @@ public function test_throwing_exception_and_requeing_when_stop_on_failure_define
...
+            $messageHandlerBuilder,
+            PollingMetadata::create('some')
+                ->setFinishWhenNoMessages(true)
+        );
+
+        $inputChannel->send(MessageBuilder::withPayload('some')->build());
+        $inputChannel->send(MessageBuilder::withPayload('some')->build());
+        $pollingConsumer->run();
+
+        $this->assertSame(0, $inputChannel->queueSize());
```
Comment (dgafka) [inline packages/Ecotone/tests/Messaging/Unit/Endpoint/Poller/PollingConsumerBuilderTest.php:448]: When you switch to In Memory channel, then in here you can run `->receive()` and assert for `null`, as everything was already consumed.

File: packages/Ecotone/tests/Messaging/Unit/Endpoint/Poller/PollingConsumerBuilderTest.php (hunk 1 above)
Comment (jlabedo) [reply packages/Ecotone/tests/Messaging/Unit/Endpoint/Poller/PollingConsumerBuilderTest.php]: I used this anonymous class to add the `queueSize` method, to check the number of messages left in the queue

File: packages/Ecotone/tests/Messaging/Unit/Endpoint/Poller/PollingConsumerBuilderTest.php (hunk 2 above)
Comment (jlabedo) [reply packages/Ecotone/tests/Messaging/Unit/Endpoint/Poller/PollingConsumerBuilderTest.php]: ok thanks

**Pull request merged**